src/webpage/src/app.rs

```diff
@@ -0,0 +1,325 @@
+use chrono::prelude::*;
+use humantime::format_duration;
+use std::collections::BTreeMap;
+
+use ewebsock::{connect, WsReceiver, WsSender};
+use url::Url;
+
+use eframe::egui::{CollapsingHeader, Context};
+use egui_plot::{Line, Plot, PlotPoints};
+
+#[derive(Clone)]
+struct FieldInfo {
+    history: Vec<(DateTime<Utc>, f64)>,
+    latest_value: f64,
+}
+
+#[derive(Clone)]
+struct MessageInfo {
+    last_sample_time: DateTime<chrono::Utc>,
+    fields: BTreeMap<String, FieldInfo>,
+}
+
+pub struct App {
+    receiver: WsReceiver,
+    sender: WsSender,
+    vehicles: BTreeMap<u8, BTreeMap<u8, BTreeMap<String, MessageInfo>>>,
+    search_query: String,
+    collapse_all: bool,
+    expand_all: bool,
+}
```

Comment:
top-to-down:

```suggestion
#[derive(Clone)]
struct MessageInfo {
    last_sample_time: DateTime<chrono::Utc>,
    fields: BTreeMap<String, FieldInfo>,
}

#[derive(Clone)]
struct FieldInfo {
    history: Vec<(DateTime<Utc>, f64)>,
    latest_value: f64,
}

pub struct App {
    receiver: WsReceiver,
    sender: WsSender,
    vehicles: BTreeMap<u8, BTreeMap<u8, BTreeMap<String, MessageInfo>>>,
    search_query: String,
    collapse_all: bool,
    expand_all: bool,
}
```

Reply:
FieldInfo is being used by MessageInfo, so it's already top down

Reply:
> FieldInfo is being used by MessageInfo, so it's already top down

that is bottom-to-top

Reply:
In this case... shouldn't it be App first ? I understand what you are saying, but it just make things confuse, since impl of the derived structures will be bigger compared with the simple structures. I think that we should follow "bottom-top" since this is kind of standard for our software development and for other languages as well.

Reply:
I could write a lot about why we should use top-to-bottom, but I won't, as it will only contribute to bikeshedding.

This repo is entirely written as top-to-bottom, so conform to it hahaha

Reply:
I went deeper on that here: https://github.com/bluerobotics/mavlink-server/pull/29#discussion_r1784107133

Reply:
You are talking/focusing on another thing there. The diff is clear: I ask for a change in the order of the declarations to be top-to-down.

Reply:
done.